### PR TITLE
Migrate from `pedantic` to `flutter_lints`

### PR DESCRIPTION
**Pull request by domesticmouse**
PTAL @johnsonmh and @clocksmith

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.1
```
Comment (guidezpl) [inline CHANGELOG.md:1]: please add date

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+### Changed
+- Migrated from `pedantic` to `flutter_lints`
```
Comment (guidezpl) [inline CHANGELOG.md:3]: nit: ```suggestion
- Migrated from `pedantic` to `flutter_lints`.
```

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Review by johnsonmh — APPROVED** [review]
LGTM

**Comment by domesticmouse** [comment]
I thought I had this working, but it turns out I can't run `flutter pub get` for this package when running on macOS Monterey. I suspect that macOS 10.15 is a little bit long in the tooth as well.

This PR is mostly working on Linux, but I have seven failing tests. I've marked these up with `skip:true` and I'd love any insights as to what I've broken.

**Comment by guidezpl** [comment]
I checked out this PR and was able to run `pub get` on macOS Monterey. There's a few places where consts can be added  by the way.
Screenshot 2021-12-08 at 15 08 45

I'm seeing the skipped tests run indefinitely rather than failing, is that what you're seeing too?

**Comment by domesticmouse** [comment]
> I checked out this PR and was able to run `pub get` on macOS Monterey. There's a few places where consts can be added  by the way.
> 
> Screenshot 2021-12-08 at 15 08 45


Here I was thinking I got them all. Darn it.

> I'm seeing the skipped tests run indefinitely rather than failing, is that what you're seeing too?

Yup, that's what I'm seeing

**Comment by domesticmouse** [comment]
I believe I have patched all the mentioned cases for `prefer_const_declarations` linter warnings.

**Comment by guidezpl** [comment]
Thanks, confirmed the tests are currently failing on `main` so this is good to go once checks finish!

**Pull request merged**